domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -513,8 +513,8 @@ func (pm *pruningManager) ArePruningPointsViolatingFinality(stagingArea *model.S
 		return false, err
 	}
 
-	for _, header := range pruningPoints {
-		blockHash := consensushashing.HeaderHash(header)
+	for i := len(pruningPoints) - 1; i >= 0; i-- {
```

Comment:
Perhaps add a comment explaining why the we must find the PP in future of virtual.F first (hence the reverse search)